moto/forecast/models.py

```diff
@@ -0,0 +1,183 @@
+from future.utils import iteritems
+import re
+from datetime import datetime
+from enum import Enum
+
+
+from boto3 import Session
+
+from moto.core import ACCOUNT_ID, BaseBackend
+from moto.core.utils import iso_8601_datetime_without_milliseconds
+from .exceptions import (
+    InvalidInputException,
+    ResourceAlreadyExistsException,
+    ResourceNotFoundException,
+    ValidationException,
+)
+
+
+class Domain(Enum):
+    RETAIL = 1
+    CUSTOM = 2
+    INVENTORY_PLANNING = 3
+    EC2_CAPACITY = 4
+    WORK_FORCE = 5
+    WEB_TRAFFIC = 6
+    METRICS = 7
+
+
+class DatasetGroup:
+    accepted_dataset_group_name_format = re.compile(r"^[a-zA-Z][a-z-A-Z0-9_]*")
+    accepted_dataset_group_arn_format = re.compile(r"^[a-zA-Z0-9\-\_\.\/\:]+$")
+    accepted_dataset_types = [
+        "INVENTORY_PLANNING",
+        "METRICS",
+        "RETAIL",
+        "EC2_CAPACITY",
+        "CUSTOM",
+        "WEB_TRAFFIC",
+        "WORK_FORCE",
+    ]
+
+    def __init__(
+        self, region_name, dataset_arns, dataset_group_name, domain, tags=None
+    ):
+        self.creation_date = iso_8601_datetime_without_milliseconds(datetime.now())
+        self.modified_date = self.creation_date
+
+        self.arn = (
+            "arn:aws:forecast:"
+            + region_name
+            + ":"
+            + str(ACCOUNT_ID)
+            + ":dataset-group/"
+            + dataset_group_name
+        )
+        self.dataset_arns = dataset_arns if dataset_arns else []
+        self.dataset_group_name = dataset_group_name
+        self.domain = domain
+        self.tags = tags
+        self._validate()
+
+    def update(self, dataset_arns):
+        self.dataset_arns = dataset_arns
+        self.last_modified_date = iso_8601_datetime_without_milliseconds(datetime.now())
+
+    def _validate(self):
+        errors = []
+
+        self._update_validation_errors(errors, self._validate_dataset_group_name())
```

Comment:
The `_update_validation_errors` helper-method** feels unnecessary. IMO it will be more readable like this:
`errors.extend(self.get_dataset_group_name_errors())`
where `get_dataset_group_name_errors()` returns a list with 0 or more errors.

Reply:
Thanks for the tip - I've refactored to accommodate your suggestion.